src/rust/src/environment.rs

```diff
@@ -11,7 +13,24 @@ impl EnvironmentReader for REnvironmentReader {
         &self,
         name: &str,
     ) -> Result<Option<Arc<dyn TableProvider>>, Box<dyn std::error::Error + Send + Sync>> {
-        // TODO: implement this
+        let classes = savvy::StringSexp(
+            savvy::eval_parse_text(format!(r#"base::get0("{name}") |> class()"#))
+                .unwrap()
+                .inner(),
+        )
+        .to_vec();
+
+        if classes.iter().any(|&s| s == "RGlareDbExecutionOutput") {
+            let sexp = savvy::Sexp(
+                savvy::eval_parse_text(format!(r#"base::get0("{name}")$.ptr"#))
+                    .unwrap()
+                    .inner(),
+            );
+            let exec = RGlareDbExecutionOutput::try_from(sexp).unwrap();
+
+            return Ok(Some(Arc::new(exec) as Arc<dyn TableProvider>));
```

Comment:
@yutannihilation Sorry for tagging, but is this a right way to get the results of executing the R code back to the Rust side?

This works well, though, but once executed, the pointer is consumed, causing a panic the second time around.
Perhaps we need to define something like a `clone()` method and make it like `base::get0("{name}")$clone()$.ptr`?

Reply:
Hmm, I don't understand what's the "right way," but probably you want `&T` instead of `T`? Then, you can probably write 

``` rust
let exec = <&RGlareDbExecutionOutput>::try_from(sexp).unwrap();
```

instead of

```rust
let exec = RGlareDbExecutionOutput::try_from(sexp).unwrap();
```

Reply:
Thanks for the quick reply!

Sorry, my intention was about how to call the R code (i.e. I didn't know how to do it with savvy as there is currently no page "Calling R Function" in the guidebook)

Reply:
Isn't you problem about how to avoid consuming the pointer stored inside the `EXTPTRSXP` object? If so, you can call `clone()` on `&RGlareDbExecutionOutput` instead of calling `$clone()` on the R object.

Reply:
> Isn't you problem about how to avoid consuming the pointer stored inside the `EXTPTRSXP` object? If so, you can call `clone()` on `&RGlareDbExecutionOutput` instead of calling `$clone()` on the R object.

Thanks, that seems to work.